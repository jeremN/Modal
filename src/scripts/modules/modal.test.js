import Modal from './index.js'

describe('Render modal', () => {
  let modal = null
  let config = {}

  beforeEach(() => {
    modal = new Modal()
    config = {
      modalTitle: 'Modal title',
      modalContent: `
        <h4>Modal content</h4>
        <input type="text" name="test">
        <button class="stan__c__button" type="button">Button</button>
      `,
      modalFooter: 'Modal footer',
    }
  })

  afterEach(() => {
    document.body.innerHTML = ''
    modal = null
    config = {}
  })

  it('should render modal correctly and add it to the DOM', () => {
    modal.onOpen(config)

    expect(document.querySelector('.js__modal')).not.toBeUndefined()
    expect(document.querySelector('.stan__c__modal__title').innerHTML).toBe('Modal title')
    expect(document.querySelector('.stan__c__modal__body').innerHTML).toContain(config.modalContent)
    expect(document.querySelector('.js__modal')).toMatchSnapshot()
  })

  it('should make body unscrollable', () => {
    modal.onOpen(config)

    expect(document.body.classList).toContain('is__noScroll')
  })

  it('should render custom template correctly', () => {
    const customConfig = {
      template: `
        <div class="stan__c__modal is__hidden"
          role="dialog"
          aria-labelledby="stan__c__modal__title"
          aria-hidden="true">
          <div class="stan__c__modal__container">
            <div class="stan__c__modal__header">
              <h2 class="stan__c__modal__title">Modal title</h2>
              <button class="stan__c__modal__close js__btnCloseModal" type="button">x</button>
            </div>
            <div class="stan__c__modal__body">
              <h3>Lorem ipsum</h3>
              <p>Dolor sit amet, consectetur adipiscing elit. Nulla velit leo, sodales ut leo pharetra, lobortis imperdiet magna.
              Fusce vel est lorem. <br> Aenean tristique, purus sed scelerisque tincidunt, ligula dolor vulputate lorem, vitae suscipit dolor purus nec lorem.</p>
            </div>
            <div class="stan__c__modal__footer">
              <button class="stan__c__button js__customModalClose">Fermer</button>
            </div>
          </div>
        </div>
      `,
    }
    modal.onOpen(customConfig)

    expect(document.querySelector('.js__modal')).toMatchSnapshot()
  })
})
