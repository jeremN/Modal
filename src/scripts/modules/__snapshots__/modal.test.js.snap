// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render modal should render custom template correctly 1`] = `
<div
  aria-hidden="false"
  aria-labelledby="stan__c__modal__title"
  class="stan__c__modal js__modal is__animated js__fadeIn"
  role="dialog"
>
  
          
  <div
    class="stan__c__modal__container is__animated js__fadeInUp"
  >
    
            
    <div
      class="stan__c__modal__header"
    >
      
              
      <h2
        class="stan__c__modal__title"
      >
        Modal title
      </h2>
      
              
      <button
        class="stan__c__modal__close js__btnCloseModal"
        type="button"
      >
        x
      </button>
      
            
    </div>
    
            
    <div
      class="stan__c__modal__body"
    >
      
              
      <h3>
        Lorem ipsum
      </h3>
      
              
      <p>
        Dolor sit amet, consectetur adipiscing elit. Nulla velit leo, sodales ut leo pharetra, lobortis imperdiet magna.
              Fusce vel est lorem. 
        <br />
         Aenean tristique, purus sed scelerisque tincidunt, ligula dolor vulputate lorem, vitae suscipit dolor purus nec lorem.
      </p>
      
            
    </div>
    
            
    <div
      class="stan__c__modal__footer"
    >
      
              
      <button
        class="stan__c__button js__customModalClose"
      >
        Fermer
      </button>
      
            
    </div>
    
          
  </div>
  
        
</div>
`;

exports[`Render modal should render modal correctly and add it to the DOM 1`] = `
<div
  aria-hidden="false"
  aria-labelledby="dialog-title"
  aria-modal="true"
  class="stan__c__modal js__modal is__animated js__fadeIn"
  id="dialog"
  role="dialog"
  tabindex="-1"
>
  
        
  <div
    class="stan__c__modal__container is__animated js__fadeInUp"
    role="document"
  >
    
          
    <div
      class="stan__c__modal__header"
    >
      
            
      <h2
        class="stan__c__modal__title"
        id="dialog-title"
      >
        Modal title
      </h2>
      
            
      <button
        aria-label="Close"
        class="stan__c__modal__close js__btnCloseModal"
        type="button"
      >
        
              
        
        <svg
          aria-hidden="true"
          height="20"
          role="img"
          viewBox="0 0 352 512"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          
          
          <path
            d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19
            0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28
            75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28
            12.28-12.28 32.19 0 44.48L109.28 256
            9.21 356.07c-12.28 12.28-12.28 32.19 0
            44.48l22.24 22.24c12.28 12.28 32.2 12.28
            44.48 0L176 322.72l100.07 100.07c12.28
            12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19
            0-44.48L242.72 256z"
          />
          
        
        </svg>
        
            
      </button>
      
          
    </div>
    
          
    <div
      class="stan__c__modal__body"
    >
      
            
        
      <h4>
        Modal content
      </h4>
      
        
      <input
        name="test"
        type="text"
      />
      
        
      <button
        class="stan__c__button"
        type="button"
      >
        Button
      </button>
      
      
          
    </div>
    

          
          
    <div
      class="stan__c__modal__footer"
    >
      
            Modal footer
          
    </div>
    
          
        
  </div>
  
      
</div>
`;
